spec/validators/currency_validator_spec.rb

```diff
@@ -0,0 +1,72 @@
+require 'rails_helper'
+
+RSpec.describe CurrencyValidator do
+  class DummyClass
+    include ActiveModel::Model
+    include ActiveModel::Validations
+
+    attr_accessor :val1, :val2, :val3
+
+    validates :val1, :val2, :val3, currency: { greater_than_or_equal_to: 0 }, allow_blank: true
+  end
+  describe 'numericality validation' do
+    let(:dummy) { DummyClass.new }
+
+    it 'raises not numeric errors on each invalid field' do
+      dummy.val1 = 'abc'
+      dummy.val2 = 'qwe'
+      expect(dummy).to be_invalid
+      expect(dummy.errors[:val1]).to eq ['is not a number']
+      expect(dummy.errors[:val2]).to eq ['is not a number']
+      expect(dummy.errors[:val3]).to be_empty
+    end
+
+    it 'errors if given negative number' do
+      dummy.val1 = '-1234'
+      dummy.val2 = '1234'
```

Comment:
I'd suggest using Faker to generate these numbers. It's possible that `1234` is a special case. It's better practice to randomly generate test numbers.
```ruby
dummy.val1 = Faker::Number.negative.to_s
dummy.val2 = Faker::Number.to_s

Reply:
no, we're testing specific Strings.